src/main/kotlin/org/jmailen/gradle/kotlinter/pluginapplier/AndroidSourceSetApplier.kt

```diff
@@ -28,8 +27,10 @@ internal object AndroidSourceSetApplier : SourceSetApplier {
             .getOrNull()
             .orEmpty()
 
+        val emptyFileTree = project.files().asFileTree
+
         return (javaSources + kotlinSources)
             .map { dir -> project.fileTree(dir) { it.include("**/*.kt") } }
-            .reduce { merged: FileTree, tree: ConfigurableFileTree -> merged + tree }
+            .fold(emptyFileTree) { merged, tree -> merged + tree }
```

Comment:
So, it seems like AGP starting from 7.3.0 doesn't include empty directories in their source sets. On 7.2.1, `javaSources` contains `src/main/java`, `kotlinSources` holds a reference to `src/main/kotlin`.
The behavior I observed on 7.3.0 was empty `javaSources` and `kotlinSources` if they did not contain any files. 
To reproduce the issue, it's enough to bump `Versions.androidTools` to anything above `7.2.1` and run existing test suite